cpp/src/binaryop/compiled/binary_ops.cuh

```diff
@@ -253,42 +257,50 @@ void for_each(rmm::cuda_stream_view stream, cudf::size_type size, Functor f)
   for_each_kernel<<<grid_size, block_size, 0, stream.value()>>>(size, std::forward<Functor&&>(f));
 }
 
-namespace detail {
-template <class BinaryOperator>
-void apply_unnested_binary_op(mutable_column_view& out,
-                              column_view const& lhs,
-                              column_view const& rhs,
-                              bool is_lhs_scalar,
-                              bool is_rhs_scalar,
-                              rmm::cuda_stream_view stream)
-{
-  auto common_dtype = get_common_type(out.type(), lhs.type(), rhs.type());
-
-  auto lhsd = column_device_view::create(lhs, stream);
-  auto rhsd = column_device_view::create(rhs, stream);
-  auto outd = mutable_column_device_view::create(out, stream);
-  // Create binop functor instance
-  auto binop_func = device_type_dispatcher<BinaryOperator>{
-    *outd, *lhsd, *rhsd, is_lhs_scalar, is_rhs_scalar, common_dtype};
-  // Execute it on every element
-  for_each(stream, out.size(), binop_func);
-}
-}  // namespace detail
-
 template <class BinaryOperator>
 void apply_binary_op(mutable_column_view& out,
                      column_view const& lhs,
                      column_view const& rhs,
                      bool is_lhs_scalar,
                      bool is_rhs_scalar,
+                     order op_order,
+                     bool flip_output,
```

Comment:
Why does this need to be added to the base dispatch function when it's only relevant to struct operations? We shouldn't need to modify _every_ op to tack on this extra arguments that are only relevant to a specific type. This should only be needed in the struct code path.

Reply:
We should not need to modify or complicate existing code paths to add this struct support. All of the stuff unique to structs should be confined to that code path.

Reply:
I had missed your followup comment when I made the most recent changes. Does the new implementation meet that requirement or should the struct specific code be pushed even further down? If I push it further down, I'll need a second layer of operator dispatching to figure out parameters to call the arguments to the struct specific code,

Reply:
The current code repeats the same problem from before of non-uniform invocation of `apply_binary_op` vs `apply_binary_op_impl`.

The struct specific arguments should be pushed down further. I don't believe it requires another level of operator dispatch, you just compute the value of those arguments as a function of the specified operator.

Reply:
Should be good to go now